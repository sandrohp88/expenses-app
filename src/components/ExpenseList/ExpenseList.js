import React from 'react'
import { connect } from 'react-redux'
import ExpenseItem from '../ExpenseItem/ExpenseItem'
import getVisibleExpenses from '../VisibleExpenses/VisibleExpenses'
export const ExpenseList = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>No expenses to show</p>
    ) : (
      <div>
        <h1>My expenses</h1>
        <ul>
          {props.expenses.map(expense => (
            <ExpenseItem key={expense.id} {...expense} />
          ))}
        </ul>
      </div>
    )}
  </div>
)

const mapStateToProps = state => ({
  expenses: getVisibleExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpenseList)
