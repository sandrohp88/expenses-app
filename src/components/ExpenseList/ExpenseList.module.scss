@import '../../goblalStyles/content-container';
@import '../../goblalStyles/visibility';

.listHeader {
  background: $off-white;
  border: 1px solid darken($off-white, 7%);
  // border-radius: 6px;
  color: $gray;
  display: flex;
  justify-content: space-between;
  padding: $m-size;
}


.noExpenses {
  display: flex;
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  color: #666666;
  padding: $s-size;
  text-decoration: none;
  transition: background 0.3s ease;
  &:hover {
    background: none;
  }
  text-align: center;
  justify-content: center;
  align-items: center;

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
    align-items: center;
  }
}
