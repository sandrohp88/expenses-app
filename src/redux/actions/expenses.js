import expensesRef from '../../firebase/firebase'
// ADD_EXPENSE
const addExpense = expense => ({
  type: 'ADD_EXPENSE',
  expense
})

// ADD_EXPENSE_ASYNC
const addExpenseAsync = (expenseData = {}) => {
  return async dispatch => {
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData
    const expense = { description, note, amount, createdAt }
    const response = await expensesRef.add(expense)
    return dispatch(addExpense({ id: response.id, ...expense }))
  }
}

// REMOVE_EXPENSE
const removeExpense = (id = '') => ({ type: 'REMOVE_EXPENSE', id })

// EDIT_EXPENSE
const editExpense = (id, updates) => ({ type: 'EDIT_EXPENSE', id, updates })

// SET_EXPENSES
const setExpenses = expenses => ({ type: 'SET_EXPENSES', expenses })

// GET_EXPENSES_ASYNC
const getExpensesAsync = () => {
  return async dispatch => {
    const response = await expensesRef.get()
    const expenses = []
    response.forEach(expense => {
      const {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
      } = expense.data()
      expenses.push({id:expense.id, description, note, amount, createdAt })
    })
    return dispatch(setExpenses(expenses))
  }
}

export {
  getExpensesAsync,
  addExpense,
  addExpenseAsync,
  removeExpense,
  editExpense,
  setExpenses
}
