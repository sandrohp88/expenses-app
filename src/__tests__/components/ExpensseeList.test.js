import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseList } from '../../components/ExpenseList/ExpenseList'
import expenses from '../../sampleData/expenses'

test('render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />)
  expect(wrapper).toMatchSnapshot()
})
test('render ExpenseList without expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />)
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('p').length).toBe(1)
})
