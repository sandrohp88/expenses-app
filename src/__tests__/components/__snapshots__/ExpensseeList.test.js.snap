// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExpenseList with expenses 1`] = `
<div>
  <div>
    <h1>
      My expenses
    </h1>
    <ul>
      <ExpenseItem
        amount={195}
        createdAt={0}
        description="Gum"
        id="1"
        key="1"
        note=""
      />
      <ExpenseItem
        amount={109500}
        createdAt={-345600000}
        description="Rent"
        id="2"
        key="2"
        note=""
      />
      <ExpenseItem
        amount={4500}
        createdAt={345600000}
        description="Credit Card"
        id="3"
        key="3"
        note=""
      />
    </ul>
  </div>
</div>
`;

exports[`render ExpenseList without expenses 1`] = `
<div>
  <p>
    No expenses to show
  </p>
</div>
`;
